drafts/SC1-4-1-link-in-text-style.md

```diff
@@ -0,0 +1,172 @@
+
+### Selector
+
+Test mode: [automatic][AUTO]
+
+1. Select each element that matches `a[href]:not(role), *[role=link]` 
+
+2. From this, take elements that meet the following requirements:
+
+  - `link.textContent` is [non-empty][NEMPTY] text
+  - Its nearest ancestor that is a [block-like element](#block-like-element) has:
+    - A different text color than the text color of the link, AND
+    - [rendered text][RNDTXT] that is not contained in the link.
+
+### Step 1: Initial state
+
+Test mode: [automatic][AUTO]
+
+Check that the link has a [distinguishing border][DSBRDR], a `background-image`, or a [distinguishing style][DSSTYL].
+
+If yes, return
+
+| Outcome  | Passed
+|----------|-----
+| Testcase | {{ page.name }}
+| ID       | {{ page.name }}-pass1
+
+Else continue with [Step 2](#step-2-link-contrast)
+
+### Step 2: Link contrast
+
+Test mode: [automatic][AUTO]
+
+Determine `color` and `background-color` of the link and it's block-like ancestor. If the background-color is transparent, use getElementsFromPoint to locate the closest underlying element that does have a `background-color` and use that value.
+
+1. C1 = contrast difference of surrounding text `color` and link `color`
+
+2. C2 = contrast difference surrounding text `background-colo` and link `background-color`
+
+If C1 or C2 is more than 3:1, continue with [Step 3](#step-3-focus-state)
+
+Else, return:
+
+| Outcome  | Failed
+|----------|-----
+| Testcase | {{ page.name }}
+| ID       | {{ page.name }}-fail1
+| Error    | The link is not sufficiently distinguishable from the surrounding text
+
+### Step 3: focus state
+
+Test mode: [automatic][AUTO]
+
+Give focus to the link.
+
+Check that the focused link has a [distinguishing border][DSBRDR], a `background-image`, or a [distinguishing style][DSSTYL].
+
+If yes, continue with [Step 4](#step-4-hover-state)
+
+Else, return:
+
+| Outcome  | Failed
+|----------|-----
+| Testcase | {{ page.name }}
+| ID       | {{ page.name }}-fail2
+| Error    | The link is not sufficiently distinguishable from the surrounding text when it receives focus.
+
+### Step 4: Hover state
+
+Test mode: [automatic][AUTO]
+
+Remove the focus from the link.
+
+Move the mouse pointer to the center of the link.
+
+Check that the focused link has a [distinguishing border][DSBRDR], a `background-image`, or a [distinguishing style][DSSTYL].
+
+If yes, return:
+
+| Outcome  | Passed
+|----------|-----
+| Testcase | {{ page.name }}
+| ID       | {{ page.name }}-pass2
+
+Else, return:
+
+| Outcome  | Failed
+|----------|-----
+| Testcase | {{ page.name }}
+| ID       | {{ page.name }}-fail3
+| Error    | The link is not sufficiently distinguishable from the surrounding text when it is hovered over with the mouse pointer.
+
+## Algorithms
+
+### Block-like element
+
+From the element, get the computed CSS value of the `display` property.
+
+Check that the value is not `inline` or starts with `inline-`.
```

Comment:
Might be too simplistic, by looking at the set of [allowed values on the `display` property](https://developer.mozilla.org/en-US/docs/Web/CSS/display). For instance, an element with a  display value of `ruby` is considered inline (I had never heard of this display property before, TBH).
Needs extra review.